### PR TITLE
Update UI to refer to Accredited Provider or Lead Partner

### PR DESCRIPTION
**Pull request by andrewscrivener**
### Context
UI changes to update references from Lead School to Lead Partner. Provider is also to be specified as Accredited Provider or Lead Partner. This is for the impending ITT reforms.

### Changes proposed in this pull request
* References to **Provider** are made specific to either **Accredited Provider** or **Lead Partner**
* References to **Lead School** are changed to **Lead Partner**
* Navigation, tab headers and other form labels use the correct term for the type of provider shown

### Guidance to review

- I haven't included the guidance pages for the time being as they're being rewritten as part of a separate ticket. 
- I've also left class/variables alone obviously to avoid conflicts.
- I'm aware that the lead schools/lead partners functionality is also being worked on behind a feature flag.

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Review by tomtrentham — COMMENTED** [review]
This is a bit of a brain-dump of thoughts, not necessarily all things for you to action. We might need to coordinate with a dev to look at overall LS/LP changes. Also for the changes in the spec files, I'm not sure we need to change those.

File: app/views/system_admin/lead_schools/index.html.erb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <%= render PageTitle::View.new(i18n_key: "lead_schools.index") %>
 
-<h1 class="govuk-heading-l">Lead schools</h1>
+<h1 class="govuk-heading-l">Lead partners</h1>
```
[the remark this answers is not in the document]
Comment (andrewscrivener) [reply app/views/system_admin/lead_schools/index.html.erb:3]: Good call. I've rolled these changes back for the time being

File: app/views/system_admin/_tab_nav.html.erb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <%= render TabNavigation::View.new(items: [
   { name: "Users", url: users_path },
-  { name: "Providers", url: providers_path },
+  { name: "Accredited providers", url: providers_path },
```
[the remark this answers is not in the document]
Comment (andrewscrivener) [reply app/views/system_admin/_tab_nav.html.erb:3]: Note your comment. I think this is fine for the navigation label but we can add some clarifying content on the page itself.

File: app/views/system_admin/providers/index.html.erb [hunk 3]
```diff
@@ -1,12 +1,12 @@
 <%= render PageTitle::View.new(i18n_key: "providers.index") %>
 
-<h1 class="govuk-heading-l">Providers</h1>
+<h1 class="govuk-heading-l">Accredited providers</h1>
```
[the remark this answers is not in the document]
Comment (andrewscrivener) [reply app/views/system_admin/providers/index.html.erb:3]: Yes, Good idea.

File: app/views/request_an_account/index.html.erb [hunk 4]
```diff
@@ -13,7 +13,7 @@
     <p class="govuk-body">You can only get a Register trainee teachers (Register) account if you work for either:</p>
     <ul class="govuk-list govuk-list--bullet">
       <li>an initial teacher training (ITT) accredited provider</li>
-      <li>a lead school involved in School Direct teacher training</li>
+      <li>a lead partner involved in School Direct teacher training</li>
```
[the remark this answers is not in the document]
Comment (andrewscrivener) [reply app/views/request_an_account/index.html.erb:16]: I think we'll leave this in for the time being as School Direct teacher training is going to need to be supported longer than expected as it's still available in Publish.

File: app/services/funding/provider_trainee_summaries_importer.rb [hunk 5]
```diff
@@ -15,7 +15,7 @@ def route_column
     end
 
     def lead_school_name_column
-      "Lead School"
+      "Lead Partner"
```
[the remark this answers is not in the document]
Comment (andrewscrivener) [reply app/services/funding/provider_trainee_summaries_importer.rb]: Updated in recent commit

File: app/models/reports/bulk_recommend_report.rb [hunk 6]
```diff
@@ -16,7 +16,7 @@ class BulkRecommendReport < TemplateClassCsv
     # additional headers
     FIRST_NAME  = "First names"
     LAST_NAME   = "Last names"
-    LEAD_SCHOOL = "Lead school"
+    LEAD_SCHOOL = "Lead partner"
```
[the remark this answers is not in the document]
Comment (andrewscrivener) [reply app/models/reports/bulk_recommend_report.rb]: I've reverted these changes for the time being as I believe they're being changed in another PR.

File: app/services/exports/funding_trainee_summary_data.rb [hunk 7]
```diff
@@ -37,7 +37,7 @@ def format_rows(trainee_summary)
             "Funding type" => funding_type_prefix(row) + trainee_summary_row_amount.payment_type,
             "Route" => row.route,
             "Course" => row.subject,
-            "Lead school" => row.lead_school_name,
+            "Lead partner" => row.lead_school_name,
```
[the remark this answers is not in the document]
Comment (andrewscrivener) [reply app/services/exports/funding_trainee_summary_data.rb]: I think these should be changed as part of a different PR if they haven't been already as there are larger implications with updating the methods. I'll revert this for the time being.

File: app/views/system_admin/lead_schools/index.html.erb [hunk 8]
```diff
@@ -1,6 +1,6 @@
 <%= render PageTitle::View.new(i18n_key: "lead_schools.index") %>
 
-<h1 class="govuk-heading-l">Lead schools</h1>
+<h1 class="govuk-heading-l">lead schools</h1>
```
Comment (Reyko) [inline app/views/system_admin/lead_schools/index.html.erb:3]: Should this be capitalized?

**Review by Reyko — APPROVED** [review]

**Comment by defong** [comment]
Closing as it is superseded by https://github.com/DFE-Digital/register-trainee-teachers/pull/4498

**Pull request closed without merging**